pygeos/predicates.py

```diff
@@ -768,3 +769,33 @@ def relate(a, b, **kwargs):
     'F0FFFF102'
     """
     return lib.relate(a, b, **kwargs)
+
+
+@multithreading_enabled
+def relate_pattern(a, b, pattern, **kwargs):
+    """
+    Returns True if the DE-9IM string code for the relationship between
+    the geometries satisfies the pattern, else False.
+
+    This function compares the DE-9IM code string for two geometries
+    against a specified pattern. If the string matches the pattern then
+    ``True`` is returned, otherwise ``False``. The pattern specified can
+    be an exact match (``0``, ``1`` or ``2``), a boolean match
+    (``T`` or ``F``), or a wildcard (``*``). For example, the pattern
+    for the `within` predicate is ``'T*F**F***'``.
```

Comment:
It looks like this is case-insensitive in GEOS.  Is that correct?  If so, is it worth noting here?

Reply:
Do you have an example of it being insensitive. 

Just trying the example of the docstring:

```
In [13]: relate_pattern(point, square, "T*F**F***")
Out[13]: True

In [14]: relate_pattern(point, square, "t*f**f***")
Out[14]: False
```

Reply:
Not sure how I saw it as case insenstive now, thought I had tested but cannot reproduce now.

Might be good to note that it is case sensitive.

Reply:
Or always force to uppercase internally?

Reply:
> Or always force to uppercase internally?

Personally, I don't see much advantage in allowing the user to also pass it as lowercase. 
(except to avoid bugs maybe where you silently get always falls results? If that's the reason, we could actually also check that it is uppercase and otherwise raise an error)